File: 01-pattern-problems/java/Pattern7.java
import java.util.Scanner;
/** 
 * @author Sourabh Ninawe
 */
class Pattern7{
    public void printPattern7(int n){
        for(int i=1;i<=n;i++){
            for(int j=1;j<=n-i;j++){
                System.out.print(" ");
            }
            for(int k=1;k<=i;k++){
                System.out.print("*");
            }
            System.out.println();
        }
    }
    public static void main(String[] args) {
        while(true){
            System.out.println("Enter the number of lines to print pattern 7 in integer and -1 to exit");
            Scanner sc = new Scanner(System.in);
            int n = sc.nextInt();
            if(n<=0)
                break;
            else{
                Pattern7 p7 = new Pattern7();
                p7.printPattern7(n);
            }
        }
    }
}

File: 01-pattern-problems/python/pattern7.py

# Author: Sourabh Ninawe
class pattern7:
    def print_pattern7_method1(self, n):
        print("pattern7:Method1")
        for i in range(1, n+1):
            for j in range(1, n-i+1):
                print(" ", end='')
            for k in range(1,i+1):
                print("*", end='')
            print()

    #shorter implementation
    def print_pattern7_method2(self, n):
        print("pattern7:Method2")
        for i in range(1, n+1):
            print(' ' * (n - i) + '*' * i)

if __name__ == "__main__":
    while True:
        try:
            n = int(input("Enter the number of lines to print pattern 7 (integer, 0 or negative to exit): "))
        except ValueError:
            print("Please enter a valid integer.")
            continue
        if n <= 0:
            break
        else:
            p7 = pattern7()
            p7.print_pattern7_method1(n)
            p7.print_pattern7_method2(n)
